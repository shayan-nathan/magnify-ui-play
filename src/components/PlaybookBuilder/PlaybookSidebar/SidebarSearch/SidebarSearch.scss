.ant-input-affix-wrapper {
  display: flex;
  align-items: center;
  margin-bottom: 16px;
}
