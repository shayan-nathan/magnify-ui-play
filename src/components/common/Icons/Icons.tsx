import IconFilter from 'assets/icons/filter.svg'
import IconSegments from 'assets/icons/segment.svg'
import IconCreateTicket from 'assets/icons/create_ticket.svg'
import IconCustomAction from 'assets/icons/custom_action.svg'
import IconIfElse from 'assets/icons/if_else.svg'
import IconMail from 'assets/icons/mail.svg'
import IconMessages from 'assets/icons/messages.svg'
import IconTime from 'assets/icons/time.svg'
import IconUpdateRecord from 'assets/icons/update_record.svg'
import IconWait from 'assets/icons/wait.svg'
import IconPen from 'assets/icons/pen.svg'
import IconFlag from 'assets/icons/flag.svg'

import { ReactComponent as IconClose } from 'assets/icons/close.svg'
import { ReactComponent as IconPenUnderlined } from 'assets/icons/pen_underlined.svg'
import { ReactComponent as IconArrowTop } from 'assets/icons/playbookBuilder/arrow_top.svg'
import { ReactComponent as IconSearch } from 'assets/icons/playbookBuilder/search.svg'
import { ReactComponent as IconCaretRight } from 'assets/icons/segmentBuilder/caret_right.svg'
import { ReactComponent as IconHome } from 'assets/icons/segmentBuilder/home.svg'
import { ReactComponent as IconArrowLeft } from 'assets/icons/segmentBuilder/arrow_left.svg'
import { ReactComponent as IconBox } from 'assets/icons/segmentBuilder/box.svg'
import { ReactComponent as IconPersons } from 'assets/icons/two_person.svg'
import { ReactComponent as IconXMark } from 'assets/icons/segmentBuilder/x_mark.svg'
import { ReactComponent as IconBookStar } from 'assets/icons/segmentBuilder/book_star.svg'
import { ReactComponent as IconGripper } from 'assets/icons/segmentBuilder/gripper.svg'
import { ReactComponent as IconBurgerMenu } from 'assets/icons/segmentBuilder/burger.svg'
import { ReactComponent as IconTrash } from 'assets/icons/playbookBuilder/trash.svg'
import { ReactComponent as DragAndDrop } from 'assets/icons/segmentBuilder/drag_and_drop_icon.svg'
import { ReactComponent as PlaceholderForward } from 'assets/icons/segmentBuilder/placeholder_forward.svg'
import { ReactComponent as DropdownArrowDown } from 'assets/icons/segmentBuilder/dropdown_arrow_down.svg'

export {
  IconFilter,
  IconSegments,
  IconCreateTicket,
  IconCustomAction,
  IconIfElse,
  IconMail,
  IconMessages,
  IconTime,
  IconUpdateRecord,
  IconWait,
  IconPen,
  IconPenUnderlined,
  IconFlag,
  IconClose,
  IconPersons,
  IconArrowTop,
  IconSearch,
  IconCaretRight,
  IconHome,
  IconArrowLeft,
  IconBox,
  IconXMark,
  IconBookStar,
  IconGripper,
  IconBurgerMenu,
  IconTrash,
  DragAndDrop,
  PlaceholderForward,
  DropdownArrowDown,
}
