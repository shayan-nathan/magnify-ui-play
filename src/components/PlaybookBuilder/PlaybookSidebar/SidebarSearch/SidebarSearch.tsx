import { FC } from 'react'
import { Input } from 'antd'

import { PlaybookActionsType } from 'components/PlaybookBuilder/PlaybookBuilderTypes'
import { filterActions } from 'components/PlaybookBuilder/Utils/filterUtils'
import { playbookBuilderOptions } from '../Settings/PlaybookBuilder.settings'
import { IconSearch } from 'components/common/Icons/Icons'

interface SidebarSearchProps {
  onChangeState: (filterOptions: PlaybookActionsType) => void
  prefix?: JSX.Element
  placeholder?: string
}

export const SidebarSearch: FC<SidebarSearchProps> = (props: SidebarSearchProps) => {
  const { onChangeState, prefix, placeholder } = props
  const icon = prefix || <IconSearch />
  const searchFilterHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
    onChangeState(filterActions(playbookBuilderOptions, event.target.value))
  }

  return (
    <Input placeholder={placeholder || 'search'} prefix={icon} onChange={searchFilterHandler} data-testid='search' />
  )
}
