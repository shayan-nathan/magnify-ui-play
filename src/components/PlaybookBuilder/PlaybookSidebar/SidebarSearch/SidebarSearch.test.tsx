import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { SidebarSearch } from './SidebarSearch'
import { IconSearch } from 'components/common/Icons/Icons'

const mockFunction = jest.fn()
// render as a asFragment, might be usefull with third-party components
const { asFragment } = render(<SidebarSearch onChangeState={mockFunction} prefix={<IconSearch />} />)
const searchComponent = screen.getByTestId('search')

describe('<SidebarSearch /> component', () => {
  it('renders correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('has the right placeholder and type', async () => {
    expect(searchComponent.getAttribute('placeholder')).toBe('search')
    expect(searchComponent.getAttribute('type')).toBe('text')
  })

  it('work typing', async () => {
    render(<SidebarSearch onChangeState={mockFunction} prefix={<IconSearch />} />)
    const searchComponent: HTMLInputElement = screen.getByTestId('search')

    // initially the field is empty
    expect(searchComponent).toHaveValue('')
    userEvent.type(searchComponent, 'Time delay')
    expect(searchComponent).toHaveValue('Time delay')
    expect(mockFunction.mock.calls.length).toBe(10)

    userEvent.clear(searchComponent)

    userEvent.type(searchComponent, 'Time{space}delay')
    expect(searchComponent).toHaveValue('Time delay')
  })
})
