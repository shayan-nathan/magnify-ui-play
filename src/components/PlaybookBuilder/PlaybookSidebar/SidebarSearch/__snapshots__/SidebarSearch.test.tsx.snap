// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarSearch /> component renders correctly 1`] = `
<DocumentFragment>
  <span
    class="ant-input-affix-wrapper"
  >
    <span
      class="ant-input-prefix"
    >
      <svg>
        search.svg
      </svg>
    </span>
    <input
      class="ant-input"
      data-testid="search"
      placeholder="search"
      type="text"
      value=""
    />
  </span>
</DocumentFragment>
`;
